app/src/main/java/com/example/quizacademytask/FlashcardStackFragment.kt

```diff
@@ -20,70 +19,61 @@ import kotlinx.coroutines.runBlocking
 class FlashcardStackFragment : Fragment() {
 
     private lateinit var pager: ViewPager2
-    private lateinit var stack: CardStack
+    private var stack: CardStack? = null
     private lateinit var db: AppDatabase
     private lateinit var cards: List<Card>
     private lateinit var toolbar: Toolbar
-    private var NUM_PAGES = 0
+    private var numPages = 0
     private lateinit var appContext: Context
-
-    private var _binding: FragmentFlashcardStackBinding? = null
-    private val binding get() = _binding!!
+    private lateinit var binding: FragmentFlashcardStackBinding
+    private lateinit var main: MainActivity
```

Comment:
ideally fragment should not have any knowledge about activity it's being displayed in. Otherwise this creates dependencies that are not needed. Here this reference can definitely be removed. Usually functions like isTablet are implemented as extensions https://kotlinlang.org/docs/extensions.html and placed in Utils.kt. You could have something like
```kotlin
Context.isTablet() {
  return context.ressources.getBoolean(R.bool.isTablet)
}
```

Then context.isTablet() would be much better readable and you don't need to retrieve reference to specific activity for this.